app/controllers/collaborators_controller.rb

```diff
@@ -0,0 +1,30 @@
+class CollaboratorsController < ApplicationController
+   before_action :authenticate_user!
+ 
+   def create
+     wiki = Wiki.find(params[:wiki_id])
+     user = User.find(params[:user_id])
+     puts user
+     collaborator = wiki.collaborators.build(user: user)
+ 
+     if collaborator.save
+       flash[:notice] = "Collaborator created with "
+     else
+       flash[:alert] = "Collaborator creation failed."
+     end
+     redirect_to wikis_path
+   end
+   
+   def destroy
+     wiki = Wiki.find(params[:wiki_id])
+     user = User.find(params[:user_id])
+     collaborator = wiki.collaborators.find(user)
+ 
```

Comment:
`collaborator = wiki.collaborators.find(user)` will try to find a `Collaborator` object with the id of `user.id`. However, `Collaborator`s are not `User`s (`Collaborator` ids are different to `User` ids) so this will produce unpredictable results.

Try finding the relevant `Collaborator` using `Collaborator.find_by(...)` with a conditions `Hash`. The conditions hash will contain the wiki id and user id submitted in the request `params`.  (`find_by` is documented in the Rails API docs (http://api.rubyonrails.org/)).